File: Lab6/Question4/src/Dot.java
public class Dot extends Shape{
    public Dot(double x, double y) {
        super(x, y);
    }

    @Override
    public double area() {
        return 0;
    }

    public static void testDot() {
        Dot dot = new Dot(1, 2);
        //Test attributes of dot
        System.out.println(dot.getX() == 1);
        System.out.println(dot.getY() == 2);
        System.out.println(dot.area() == 0);
    }
}

File: Lab8/Dot.kt
class Dot : Shape {
    constructor(x: Double, y: Double) : super(x, y)

    override fun area(): Double {
        return 0.0
    }

    override fun resize(newSize: Double) {
        throw CannotResizeException("Cannot resize a dot!")
    }

    companion object {
        fun testDot() {
            val dot = Dot(2.0, 1.0)
            println(dot.area() == 0.0)
            println(dot.getX() == 2.0)
            println(dot.getY() == 1.0)
            dot.move(1.0, 2.0)
            println(dot.getX() == 3.0)
            println(dot.getY() == 3.0)
            try {
                dot.resize(13.0)
            }
            catch (e: CannotResizeException) {
                println("Error: ${e.message}")
            }
        }
    }
}